### ISSUE TITLE
[Request] Can `Model::destroy(IDs)` also delete related models specified by `with(relations)`

### ISSUE THREAD

**Issue by vHeemstra** [laravel/ideas#1525]
Is it possible to let a call to a model's `destroy` method also delete optionally specified related models (using `with(...relationship names...)`?

For example: `Post::with(['comments', 'images'])->destroy([1, 2, 5]);`

**Comment by sisve** [comment]
Are you thinking of cascading deletes on foreign keys?

https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html

**Comment by vHeemstra** [comment]
No, more like an optional way to also delete related models. So only the related models of the specified relationships (using the `with()` function) would be deleted. (And none if no `with()` call was used before the `->delete()` or `->destroy()`.)

**Comment by vHeemstra** [comment]
Of course there are multiple ways to delete/detach related models, but I'm suggesting using the `with()` function as an elegant way to do this (without having to retrieve all the models and related models first).

See also my [question on StackOverflow](https://stackoverflow.com/questions/54740191/laravel-how-to-delete-multiple-models-including-optional-relationships).

**Comment by Sladewill** [comment]
Theres also the consideration that the MySQL foreign key way of doing it doesn't handle soft deletes.
